anndata/_core/views.py

```diff
@@ -198,6 +199,78 @@ def as_view_zappy(z, view_args):
     return z
 
 
+try:
+    from ..compat import awkward as ak
+    import weakref
+
+    # Registry to store weak references from AwkwardArrayViews to their parent AnnData container
+    _registry = weakref.WeakValueDictionary()
+    _PARAM_NAME = "_view_args"
+
+    class AwkwardArrayView(_ViewMixin, AwkArray):
+        @property
+        def _view_args(self):
+            """Override _view_args to retrieve the values from awkward arrays parameters.
+
+            Awkward arrays cannot be subclassed like other python objects. Instead subclasses need
+            to be attached as "behavior". These "behaviors" cannot take any additional parameters (as we do
+            for other data types to store `_view_args`). Therefore, we need to store `_view_args` using awkward's
+            parameter mechanism. These parameters need to be json-serializable, which is why we can't store
+            ElementRef directly, but need to replace the reference to the parent AnnDataView container with a weak
+            reference.
+            """
+            parent_key, attrname, keys = self.layout.parameter(_PARAM_NAME)
+            if parent_key is None or attrname is None or keys is None:
+                raise KeyError(
+                    "AwkwardArrayView does not hold reference to original AnnData object."
+                )
+            else:
+                try:
```

Comment:
@grst. these aren't being hit by tests, but I'm also not totally sure when these would come up.

Is this like:

```python
view_of_awk = adata[:2].obsm["awk"]
del adata
view_of_awk[:] = 1
```

Reply:
This is more of a precaution. I can turn it into an AssertionError or even remove it?

Reply:
I guess I would mainly like this code to have coverage so there's and example of when it's used.

It will be hard to tell if this code can be removed in future if we don't have an example. If it's never used, then I think we can remove it now.